engine/src/main/java/org/pentaho/di/trans/ael/websocket/MessageEventService.java

```diff
@@ -176,10 +177,14 @@ public int compare( Message o1, Message o2 ) {
           //others messages from the Daemon to Client are PDIEvent <RemoteSource, ?>
           RemoteSource keyRemoteSource = ( (RemoteSource) ( (PDIEvent) o1 ).getSource() );
           RemoteSource remoteSource = ( (RemoteSource) ( (PDIEvent) o2 ).getSource() );
+
+          // Note: `== ModelType.TRANSFORMATION` is to allow all transformation type logging to come through to the
+          // client specifically for AEL subtrans/subjob logs
           if ( keyRemoteSource.getModelType() == remoteSource.getModelType()
```

Comment:
Is there any refactoring we can do to pull these conditions into a method with a name to help understand what we're checking?

Reply:
Can do.

Reply:
I personally don't know all the use cases. I would have to discuss with @mkambol to give them all appropriate names.